src/sql/base/parts/editableDropdown/browser/dropdown.ts

```diff
@@ -146,6 +146,16 @@ export class Dropdown extends Disposable implements IListVirtualDelegate<string>
 			}
 		}));
 
+		/*
+			This event listener is intended to commit the database selection that was manually entered into the
+			editable drop down, so that Azure Data Studio's database selector can behave just like the one in SSMS.
+		*/
+		this._register(DOM.addDisposableListener(this._input.inputElement, DOM.EventType.BLUR, () => {
+			if (this.input.validate() === undefined) {
+				this._onValueChange.fire(this._input.value);
```

Comment:
We already have [fireOnTextChange](https://github.com/microsoft/azuredatastudio/blob/main/src/sql/base/parts/editableDropdown/browser/dropdown.ts#L77) that controls whether the onValueChange event is fired whenever the text changes. Did you look into using that?

In general we should be careful about making changes like this to core components. Dropdowns are used all across the app and changing the behavior like this could result in issues in other parts of the app. A change like this doesn't necessarily seem like something all consumers would want to have be the default behavior.

Reply:
@lewis-sanchez have you tried to set the fireOnTextChange option to true?

Reply:
are you not using that because it will be fired multiple times?

Reply:
Yes, he tried it. And I don't think that's the correct answer because yes, it would be fired every single time the text changed. We only really care about getting the value when the user either selects a new value or tabs out of the select box so onBlur seems like the better option to use.

Reply:
agreed, using onBlur sounds good to me.

Reply:
@alanrenmsft when I set the fireOnTextChange option to true, that broke the filtering aspect of the db selector drop down because it would show all of them. With the onBlur approach that isn't the case and filtering works as expected.